### DISCUSSION TITLE
Testing with cookie authentication

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#709] [Q&A] [answered]
I've seen in https://github.com/fastapi-users/fastapi-users/discussions/671 that we can use a fixture to test protected routes, but I also wanted to check at a lower level that cookies are set properly. Unfortunately, the following code doesn't work:
```python
def test_auth(client):
    # who am I? unauthorized
    response = client.get("/me")
    assert response.status_code == 401  
    # register user
    register_body = {
        "email": "user@example.com",
        "password": "string",
        "is_active": True,
        "is_superuser": False,
        "is_verified": False,
    }
    response = client.post("/register", json=register_body)
    assert response.status_code == 201 or response.status_code == 400
    # login with registered user
    login_body = {
        "username": "user@example.com",
        "password": "string"
    }
    assert "fastapiusersauth" not in client.cookies
    response = client.post("/login", data=login_body)
    assert "fastapiusersauth" in client.cookies
    # who am I?
    response = client.get("/me")
    assert response.status_code != 401
```
The last assertion is false, and we get an unauthorized response code, even though the `fastapiusersauth` cookie has been set.
I know it's using `requests` under the hood, could it be that the cookies are not sent?

**Comment by frankie567** [comment, marked as answer]
Hmm, I think I know what's going on here. Is your cookie authentication backend set with `secure=True` flag? If so, the `requests` library will strictly honor it and won't send it in this context without SSL (see https://github.com/tiangolo/fastapi/issues/3339#issuecomment-857669759)

What you can do is to set the `secure` flag dynamically using an environment variable to disable it in local.

**Comment by davidbrochart** [comment]
That was it, thanks a lot!
